bin/parse-layers

```diff
@@ -0,0 +1,388 @@
+                references_dict[member[ref_key]].append(member)
+        members_with_aliases = []
+        for key in references_dict:
+            aliases = []
+            if len(references_dict[key]) > 1:
+                for j in range(len(references_dict[key])):
+                    aliases.append(references_dict[key][j]['name'])
+                references_dict[key][0]['name'], references_dict[key][0]['aliases'] = self.remove_longest_name(aliases)
+            members_with_aliases.append(references_dict[key][0])
+        return members_with_aliases
+
+    def _get_call_arguments(self, member):
+        mros = member.mro()
+        call_method = None
+        for superclass in mros:
+            if superclass.__dict__.get('__call__'):
+                call_method = superclass.__dict__.get('__call__')
+                break
+
+            if superclass.__dict__.get('call'):
+                call_method = superclass.__dict__.get('call')
+                break
+
+        if call_method:
+            call_args = self._get_fn_arguments(call_method)
+            for arg in call_args:
+                if arg.get('name') == 'args':
+                    arg['name'] = 'inputs'
+                    arg['default'] = None
+            return call_args
+
+    def _get_fn_arguments(self, member):
+        if inspect.isfunction(member):
+            params_list = []
+            params = inspect.signature(member).parameters
+            for param_name, param in params.items():
+                if param_name == 'kwargs':
+                    continue
+                type_ = self._infer_argument_type(param_name, param)
+                if param.default is None or isinstance(param.default, bool):
+                    default = str(param.default)
+                else:
+                    default = None if param.default is param.empty else param.default
+
+                if isinstance(default, tf.python.framework.dtypes.DType):
+                    default = str(default)
+
+                params_list.append({
+                    'name': param_name,
+                    'default': default
+                })
+                if type_:
+                    params_list[-1].update({'type': type_})
+
+            if len(params_list) and inspect.isclass(member):
+                params_list.insert(0, {'name': 'self', 'default': None})
+            elif inspect.isclass(member):
+                params_list = None
+
+            return params_list
+
+    @staticmethod
+    def _infer_argument_type(param_name, param):
+        if type(param.default) in PRIMITIVE_TYPE_MAPS:
+            return PRIMITIVE_TYPE_MAPS[type(param.default)]
+
+    @staticmethod
+    def delete_key(list_of_dict, key='reference'):
+        for member in list_of_dict:
+            member.pop(key)
+
+    @staticmethod
+    def is_helper_type(member):
+        return member in SERIALIZATION_HELPERS or member in INIT_HELPERS
+
+    @staticmethod
+    def remove_longest_name(list_of_names):
```

Comment:
Is there a better way to determine the default name to use when imported?

Reply:
This is handled in a better way to use the standard class names for layers.